### PR TITLE
Feature/update settings button

### LINKED ISSUE #285: Naviguation bar: update settings button
## Description

When the user is successful connected. 

Indicate in the nav bar, that he is connected with its initial.

To implements:

- Update Button to dropdown button with colored text
- Retrieve user information from OIDC, add configuration to make mapping between provider information and our application
- Re-design menu

Not to implements:
- Missing menu action (logout, account settings)

example of wanted configuration:
```js
{
  "OIDC": [{
    "userMapping": {
      "firstname": "profil.user.firstname",
      "lastname": "profil.user.lastname",
      "email": "profil.user.email",
      "userId": "profil.user.id",
    },
  }]
}
```

Image

### PR DESCRIPTION
**Pull request by k-y0u**
resolves #285

### CODE REVIEW

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 1]
```diff
@@ -1,41 +1,57 @@
...
+        {{ `${user.value.firstname.at(0)}${user.value.lastname.at(0)}` }}
+      </q-avatar>
+      <q-avatar
+        v-else
+        color="primary"
+        text-color="white"
+        icon="fa-solid fa-user"
+        size="md"
+        font-size="12px"
+        title="Unknown User"
```
Comment (justinehell) [inline src/components/menu/ModelizerSettingsMenu.vue:27]: use of i18n

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 2]
```diff
@@ -1,41 +1,57 @@
...
+    <template #label>
+      <q-avatar
+        v-if="user?.firstname && user?.lastname"
+        color="primary"
+        text-color="white"
+        size="md"
+        font-size="12px"
+        :title="`${user.firstname} ${user.lastname}`"
+      >
+        {{ `${user.value.firstname.at(0)}${user.value.lastname.at(0)}` }}
```
Comment (justinehell) [inline src/components/menu/ModelizerSettingsMenu.vue:18]: why not using a computed to improve readability ?

File: README.md [hunk 3]
```diff
@@ -181,6 +188,7 @@ Example with `Google` and `Github` providers.
 
 - the `name` is the name of the provider. It is required.
 - the `icon` is the icon of the provider. It is required, must be named `/[OIDC[x].name]/i.svg` and be placed inside `public/provider` folder.
+- the `userMapping` is the map between needed user parameters and wher to find them. It is required.
```
Comment (justinehell) [inline README.md:191]: wher -> where

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/menu/ModelizerSettingsMenu.vue (hunk 1 above)
Comment (k-y0u) [reply src/components/menu/ModelizerSettingsMenu.vue]: Done

File: README.md (hunk 3 above)
Comment (k-y0u) [reply README.md]: Done

File: src/composables/UserAuthentication.js [hunk 4]
```diff
@@ -73,6 +78,11 @@ export function completeSilentLogin() {
  * to fetch the user's information from the authentication provider.
  * @return {Promise<Object>} Promise with the current user on success otherwise an error.
  */
-export function getUser() {
-  return userManager.getUser();
+export async function getUser() {
+  return userManager.getUser().then((user) => new User({
```
Comment (justinehell) [inline src/composables/UserAuthentication.js:82]: what if userManager is null ?
image

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 5]
```diff
@@ -1,41 +1,57 @@
...
+        {{ userInitials }}
+      </q-avatar>
+      <q-avatar
+        v-else
+        color="primary"
+        text-color="white"
+        icon="fa-solid fa-user"
+        size="md"
+        font-size="12px"
+        :title="$t(page.modelizer.settings.user.unknown)"
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/components/menu/ModelizerSettingsMenu.vue]: done

File: src/models/User.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
+/**
+ * User class.
+ */
+class User {
+  /**
+   * Default constructor.
+   * @param {object} [props] - Object that contains all properties to set.
+   * @param {string} [props.firstname] - Template to retrive firstname in user.
```
Comment (Zorin95670) [inline src/models/User.js:8]: for all, not a template its the user firstname.

File: src/composables/UserAuthentication.js [hunk 7]
```diff
@@ -73,6 +78,11 @@ export function completeSilentLogin() {
  * to fetch the user's information from the authentication provider.
  * @returns {Promise<object>} Promise with the current user on success otherwise an error.
  */
-export function getUser() {
-  return userManager.getUser();
+export async function getUser() {
+  return userManager?.getUser().then((user) => new User({
+    firstname: nunjucks.renderString(`{{ ${user?.profile?.firstname} }}`, user),
```
Comment (Zorin95670) [inline src/composables/UserAuthentication.js:83]:
```js
firstname: nunjucks.renderString(`{{ ${userAuthentication.userMapping.firstname} }}`, user),
```
for all

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 8]
```diff
@@ -73,6 +91,9 @@ const menuItems = computed(() => [
     visible: true,
   },
 ]);
+const userInitials = computed(() => (user.value?.firstname && user.value?.lastname
+  ? `${user.value.firstname.at(0)}${user.value.lastname.at(0)}`
+  : ''));
```
Comment (Zorin95670) [inline src/components/menu/ModelizerSettingsMenu.vue:96]: : null at the end

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 9]
```diff
@@ -1,41 +1,57 @@
...
+        text-color="white"
+        size="md"
+        font-size="12px"
+        :title="`${user.firstname} ${user.lastname}`"
+      >
+        {{ userInitials }}
+      </q-avatar>
+      <q-avatar
+        v-else
+        color="primary"
```
Comment (Zorin95670) [inline src/components/menu/ModelizerSettingsMenu.vue:22]: grey

File: README.md [hunk 10]
```diff
@@ -213,7 +227,7 @@ The `config` object might change from one provider to another. Here is an exampl
             "silent_redirect_uri": "https://your-app.com/silent-refresh",
             "token_endpoint": "https://github.com/login/oauth/access_token/",
             "token_type": "bearer",
-            "client_secret": "your-client-secret",
```
Comment (Zorin95670) [inline README.md:216]: Update config

File: changelog.md [hunk 11]
```diff
@@ -42,6 +42,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Models page:
   * Rework on tag display, use tag category for all diagrams in table.
   * Rework on tag display, use tag language for plugin selection on create diagram from template.
+  * Update ModelizerSettingsMenu to display user info.
```
Comment (Zorin95670) [inline changelog.md:45]: Update naviguation bar to display user info

File: src/models/User.js (hunk 6 above)
Comment (k-y0u) [reply src/models/User.js]: done

File: src/components/menu/ModelizerSettingsMenu.vue (hunk 8 above)
Comment (k-y0u) [reply src/components/menu/ModelizerSettingsMenu.vue]: done

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 12]
```diff
@@ -1,41 +1,57 @@
...
+    <template #label>
+      <q-avatar
+        v-if="user?.firstname && user?.lastname"
+        color="primary"
+        text-color="white"
+        size="md"
+        font-size="12px"
+        :title="`${user.firstname} ${user.lastname}`"
+      >
+        {{ userInitials }}
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/components/menu/ModelizerSettingsMenu.vue:18]: nope

File: src/components/menu/ModelizerSettingsMenu.vue (hunk 9 above)
Comment (k-y0u) [reply src/components/menu/ModelizerSettingsMenu.vue]: done

File: README.md (hunk 10 above)
Comment (k-y0u) [reply README.md:216]: done

File: changelog.md (hunk 11 above)
Comment (k-y0u) [reply changelog.md]: done

File: src/components/menu/ModelizerSettingsMenu.vue (hunk 8 above)
Comment (Zorin95670) [reply src/components/menu/ModelizerSettingsMenu.vue]: You simplify the code, but I prefere you to keep previous version with this changes:
```
const userInitials = computed(() => (user.value?.firstname && user.value?.lastname
  ? `${user.value.firstname.at(0)}${user.value.lastname.at(0)}`
  : t('page.modelizer.settings.user.unknownInitials')));
```

File: src/components/menu/ModelizerSettingsMenu.vue (hunk 12 above)
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/menu/ModelizerSettingsMenu.vue:18]: answer in another comment

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 13]
```diff
@@ -73,6 +91,7 @@ const menuItems = computed(() => [
     visible: true,
   },
 ]);
+const userInitials = computed(() => `${user.value.firstname?.at(0)}${user.value.lastname?.at(0)}`);
```
Comment (Zorin95670) [inline src/components/menu/ModelizerSettingsMenu.vue:94]: add toUpperCase at the end

File: src/components/menu/ModelizerSettingsMenu.vue (hunk 8 above)
Comment (Zorin95670) [reply src/components/menu/ModelizerSettingsMenu.vue]: My bad, I don't see the v-if.

File: src/components/menu/ModelizerSettingsMenu.vue (hunk 13 above)
Comment (k-y0u) [reply src/components/menu/ModelizerSettingsMenu.vue]: done

File: src/components/menu/ModelizerSettingsMenu.vue (hunk 8 above)
Comment (k-y0u) [reply src/components/menu/ModelizerSettingsMenu.vue]: nope

**Review by Zorin95670 — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**